controllers/machineset_controller.go

```diff
@@ -399,14 +400,14 @@ func (r *MachineSetReconciler) syncReplicas(ctx context.Context, ms *clusterv1.M
 			}(machine)
 		}
 		wg.Wait()
+		close(errCh)
 
-		select {
-		case err := <-errCh:
-			// all errors have been reported before and they're likely to be the same, so we'll only return the first one we hit.
-			if err != nil {
-				return err
-			}
-		default:
+		var errs []error
+		for err := range errCh {
+			errs = append(errs, err)
+		}
+		if len(errs) > 0 {
```

Comment:
This length check is already handled inside `NewAggregate()` so it's not necessary here.

Reply:
If there is no error, we need to call r.waitForMachineDeletion().
So this check is needed.

Reply:
My bad, didn't read down below. If you have
```go
if len(errs) > 0 {
  return kerrors.NewAggregate(errs)
}

return nil
```

then my comment would be appropriate. Sorry for the confusion.